ibis/sql/postgres/tests/test_udf.py

```diff
@@ -0,0 +1,173 @@
+#     pl/python tests.
+#     TODO: update Windows pipeline to exclude postgres_extensions
+#     TODO: remove postgis marker below once Windows pipeline updated
+pytestmark = [
+    pytest.mark.postgresql,
+    pytest.mark.postgis,
+    pytest.mark.postgres_extensions,
+]
+
+# Database setup (tables and UDFs)
+
+
+test_schema = None
+
+
+def gen_schema_name(basename='test'):
+    """Generate a random alpha string starting with 'test_' to be used as a
+    test schema name"""
+    schema_name = '{}_{}'.format(
+        basename,
+        ''.join([random.choice(string.ascii_lowercase) for i in range(6)])
+    )
+    return schema_name
+
+
+def create_test_schema(connection: PostgreSQLClient, max_tries=20):
+    """Create a random test schema, necessary to allow for parallel pytest
+    testing"""
+    schema_exists = True
+    tries = 0
+    while schema_exists and tries < max_tries:
+        new_name = gen_schema_name()
+        sql_check_schema = """SELECT count(schema_name) as n_recs
+        FROM information_schema.schemata
+        WHERE lower(schema_name) = lower('{}');""".format(new_name)
+        df_result = pd.read_sql(sql_check_schema, connection.con)
+        schema_exists = df_result['n_recs'].iloc[0] > 0
+        tries += 1
+    assert tries > 0
+    global test_schema
+    test_schema = new_name
+    connection.con.execute("CREATE SCHEMA {};".format(new_name))
+
+
+table_name = 'udf_test_users'
+
+sql_table_setup = """DROP TABLE IF EXISTS {schema}.{table_name};
+CREATE TABLE {schema}.{table_name} (
+    user_id integer,
+    user_name varchar,
+    name_length integer
+);
+INSERT INTO {schema}.{table_name} VALUES
+(1, 'Raj', 3),
+(2, 'Judy', 4),
+(3, 'Jonathan', 8)
+;
+"""
+
+sql_define_py_udf = """CREATE OR REPLACE FUNCTION {schema}.pylen(x varchar)
+RETURNS integer
+LANGUAGE plpythonu
+AS
+$$
+return len(x)
+$$;"""
+
+sql_define_udf = """CREATE OR REPLACE FUNCTION {schema}.custom_len(x varchar)
+RETURNS integer
+LANGUAGE SQL
+AS
+$$
+SELECT length(x);
+$$;"""
+
+
+@pytest.fixture
+def con_for_udf(con):
+    create_test_schema(con)
+    con.con.execute(
+        sql_table_setup.format(table_name=table_name, schema=test_schema)
+    )
+    con.con.execute(sql_define_udf.format(schema=test_schema))
+    con.con.execute(sql_define_py_udf.format(schema=test_schema))
+    yield con
+    # teardown
+    con.con.execute("DROP SCHEMA IF EXISTS {} CASCADE".format(test_schema))
+
+
+@pytest.fixture
+def table(con_for_udf):
+    return con_for_udf.table(table_name, schema=test_schema)
+
+# Tests
+
+
+def test_sql_length_udf_worked(table):
+    """Test creating ibis UDF object based on existing UDF in the database"""
+    # Create ibis UDF objects referring to UDFs already created in the database
+    custom_length_udf = ibis.sql.postgres.udf.api.existing_udf(
+        'custom_len',
+        input_types=[ibis.expr.datatypes.String()],
+        output_type=ibis.expr.datatypes.Integer(),
```

Comment:
`Integer` is an abstract class that is the base of all other integer types. This should be something like `dt.int64`, `dt.int32`, etc.